src/ypspur_ros.cpp

```diff
@@ -1005,10 +1014,11 @@ class YpspurRosNode
         }
         pubs_["joint"].publish(joint);
 
+        tf2::Quaternion quat_tf;
         for (unsigned int i = 0; i < joints_.size(); i++)
         {
-          joint_trans[i].transform.rotation =
-              tf::createQuaternionMsgFromYaw(joint.position[i]);
+          quat_tf.setRPY(0, 0, joint.position[i]);
```

Comment:
```suggestion
          const tf2::Quaternion quat_tf(0, 0, joint.position[i]);
```
api ref: http://docs.ros.org/en/jade/api/tf2/html/classtf2_1_1Quaternion.html#a29cf4deb42a3a1867a066b9f74578bf6

Reply:
This method seems to be deprecated http://docs.ros.org/en/noetic/api/tf2/html/classtf2_1_1Quaternion.html#a3e763411f10c67bd23bbfcc1b880793f. We can use this one though: http://docs.ros.org/en/noetic/api/tf2/html/classtf2_1_1Quaternion.html#ae23b2f32c74adb17fceb71187346f3af

Reply:
```
trans.transform.rotation = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0, 0, 1), msg->position[i]));
```

Reply:
Done in f8fbe7eca99856408d717dfb69097cc08c6fd596